### PR TITLE
refactor interest rate model

### PR DESCRIPTION
**Pull request by yz89**
- Add `APR` struct to make the rate model more clear.
- Add set_rate_model call and tests

### CODE REVIEW

File: pallets/loans/src/lib.rs [hunk 1]
```diff
@@ -82,8 +83,9 @@ pub mod module {
         #[pallet::constant]
         type PalletId: Get<PalletId>;
 
-        /// The origin which can add/reduce reserves.
-        type ReserveOrigin: EnsureOrigin<Self::Origin>;
+        /// The origin which can update rate model, liquidate incentive and
+        /// add/reduce reserves. Root can always do this.
+        type UpdateOrigin: EnsureOrigin<Self::Origin>;
```
Comment (ayushmishra2005) [inline pallets/loans/src/lib.rs:88]: I think ReserveOrigin suits more here.

File: pallets/loans/src/lib.rs [hunk 2]
```diff
@@ -550,6 +559,25 @@ pub mod module {
             Ok(().into())
         }
 
+        /// Update the interest rate model for a given asset.
+        #[pallet::weight(10_000)]
+        #[transactional]
+        pub fn set_rate_model(
```
Comment (ayushmishra2005) [inline pallets/loans/src/lib.rs:565]: Can we add benchmarking and weight of this new dispatchable function?

File: node/parallel-dev/src/chain_spec.rs [hunk 3]
```diff
@@ -187,12 +187,12 @@ fn testnet_genesis(
...
-            borrow_index: Rate::one(),                                  // 1
-            exchange_rate: Rate::saturating_from_rational(2, 100),      // 0.02
-            base_rate_per_year: Rate::saturating_from_rational(2, 100), // 0.02
-            multiplier_per_year: Multiplier::saturating_from_rational(1, 10), // 0.1
-            jump_multiplier_per_year: Multiplier::saturating_from_rational(11, 10), // 1.1
-            kink: Ratio::from_percent(80),                              // 0.8
+            borrow_index: Rate::one(),                             // 1
+            exchange_rate: Rate::saturating_from_rational(2, 100), // 0.02
+            base_rate: Rate::saturating_from_rational(2, 100),     // 2%
+            kink_rate: Rate::saturating_from_rational(10, 100),    // 10%
```
Comment (GopherJ) [inline node/parallel-dev/src/chain_spec.rs:193]: `kink_rate` isn't a good name as this value is between `0` and `1`, ratio is better

**Review by GopherJ — APPROVED** [review]

**Comment by ayushmishra2005** [comment]
@yz89 Can you rebase with master?

**Comment by yz89** [comment]
It needn't rebase with the master if no conflicts, because we will use squash and merge. @ayushmishra2005

**Review by ayushmishra2005 — APPROVED** [review]

**Pull request merged**